src/main/java/htsjdk/samtools/fastq/FastqRecord.java

```diff
@@ -88,37 +194,33 @@ public boolean equals(Object obj) {
         if (getClass() != obj.getClass())
             return false;
         FastqRecord other = (FastqRecord) obj;
-        if (seqLine == null) {
-            if (other.seqLine != null)
+        if (readString == null) {
+            if (other.readString != null)
                 return false;
-        } else if (!seqLine.equals(other.seqLine))
+        } else if (!readString.equals(other.readString))
             return false;
-        if (qualHeaderPrefix == null) {
-            if (other.qualHeaderPrefix != null)
+        if (qualityHeader == null) {
+            if (other.qualityHeader != null)
                 return false;
-        } else if (!qualHeaderPrefix.equals(other.qualHeaderPrefix))
+        } else if (!qualityHeader.equals(other.qualityHeader))
             return false;
-        if (qualLine == null) {
-            if (other.qualLine != null)
+        if (baseQualityString == null) {
+            if (other.baseQualityString != null)
                 return false;
-        } else if (!qualLine.equals(other.qualLine))
+        } else if (!baseQualityString.equals(other.baseQualityString))
             return false;
-        if (seqHeaderPrefix == null) {
-            if (other.seqHeaderPrefix != null)
+        if (readName == null) {
+            if (other.readName != null)
                 return false;
-        } else if (!seqHeaderPrefix.equals(other.seqHeaderPrefix))
+        } else if (!readName.equals(other.readName))
             return false;
-        
+
         return true;
     }
-    
+
+    /** Simple toString() that gives a read name and length */
     @Override
     public String toString() {
```

Comment:
It is discourage to use `toString()` to other purposes that is not developmental, but it is true that some code may rely on this. I changed it and point to the `toFastQString`, but I would rather prefer to add a comment saying that this is discouraged and that this will change at some point (if we set a date, will be nice). What do you think, @lbergelson?